### PR TITLE
Implementing ArtFilter class

### PR DESCRIPTION
**Pull request by AnnaBanani**
Implementing ArtFilter class

### CODE REVIEW

File: TheMET/TheMET/ArtFilter/ArtFilter.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+//
+//  Created by Анна Ситникова on 10/08/2023.
+//
+
+import Foundation
+import UIKit
+
+class ArtFilter {
+    func filter(arts: [Art], searchText: String?) -> [Art] {
+        var filteredArts:[Art] = []
```
Comment (Ssimboss) [inline TheMET/TheMET/ArtFilter/ArtFilter.swift:13]: This array is better to instantiate after the guard-check

File: TheMET/TheMET/ArtFilter/ArtFilter.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        for art in arts {
+            if self.filter(artProperty: art.artistDisplayName, searchText: searchText) ||
+                self.filter(artProperty: art.department, searchText: searchText) ||
+                self.filter(artProperty: art.medium, searchText: searchText) ||
+                self.filter(artProperty: art.objectName, searchText: searchText) ||
+                self.filter(artProperty: art.title, searchText: searchText) {
+                guard !filteredArts.contains(where: { filteredArt in
+                    filteredArt.objectID == art.objectID
+                }) else {
+                    return filteredArts
```
Comment (Ssimboss) [inline TheMET/TheMET/ArtFilter/ArtFilter.swift:26]: You're executing `return` in the middle of the `for`-cycle. It means some elements of the original array are nit checked yet and won't be included to the result

File: TheMET/TheMET/ArtFilter/ArtFilter.swift [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                }
+                filteredArts.append(art)
+            }
+        }
+        return filteredArts
+    }
+    
+    private func filter(artProperty: String?, searchText: String) -> Bool {
+        if let artProperty = artProperty,
+           let _ = artProperty.range(of: searchText, options: .caseInsensitive) {
```
Comment (Ssimboss) [inline TheMET/TheMET/ArtFilter/ArtFilter.swift:36]: there is no need to use `let  _`, you can just check that the result of `range` is not `nil`

File: TheMET/TheMET/ArtFilter/ArtFilter.swift [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        guard let searchText = searchText else {
+            return arts
+        }
+        for art in arts {
+            if self.filter(artProperty: art.artistDisplayName, searchText: searchText) ||
+                self.filter(artProperty: art.department, searchText: searchText) ||
+                self.filter(artProperty: art.medium, searchText: searchText) ||
+                self.filter(artProperty: art.objectName, searchText: searchText) ||
+                self.filter(artProperty: art.title, searchText: searchText) {
+                guard !filteredArts.contains(where: { filteredArt in
```
Comment (Ssimboss) [inline TheMET/TheMET/ArtFilter/ArtFilter.swift:23]: Why do you need this check?

**Review by Ssimboss — CHANGES_REQUESTED** [review]

File: TheMET/TheMET/ArtFilter/ArtFilter.swift (hunk 1 above)
Comment (AnnaBanani) [reply TheMET/TheMET/ArtFilter/ArtFilter.swift:13]: +

File: TheMET/TheMET/ArtFilter/ArtFilter.swift (hunk 2 above)
Comment (AnnaBanani) [reply TheMET/TheMET/ArtFilter/ArtFilter.swift:26]: +

File: TheMET/TheMET/ArtFilter/ArtFilter.swift (hunk 3 above)
Comment (AnnaBanani) [reply TheMET/TheMET/ArtFilter/ArtFilter.swift:36]: +

File: TheMET/TheMET/ArtFilter/ArtFilter.swift (hunk 4 above)
Comment (AnnaBanani) [reply TheMET/TheMET/ArtFilter/ArtFilter.swift:23]: +

**Review by Ssimboss — APPROVED** [review]

**Pull request merged**